modules/core/src/test/java/org/apache/ignite/internal/processors/diagnostic/DiagnosticProcessorTest.java

```diff
@@ -167,24 +192,237 @@ public void testWalDirs() throws Exception {
+        assertTrue(new File(partitionsPath, partFileName).exists());
+    }
+
+    /**
+     * Validates that {@code <baseDumpDir>/wal} directory exists and contains WAL segments.
+     */
+    private static void validateWalDir(File baseDumpDir) {
+        File walPath = new File(baseDumpDir, "wal");
+        assertTrue(walPath.exists());
+        assertTrue(walPath.isDirectory());
+
+        assertEquals(1, walPath.list().length);
+        assertTrue(new File(walPath, FileDescriptor.fileName(0)).exists());
+    }
+
+    /**
+     * Validates that {@code <baseDumpDir>/binary_meta} directory exists and contains binary meta files.
+     */
+    private static void validateBinaryMetaDir(File baseDumpDir) {
+        // "dump/binary_meta" directory should contain a single file.
+        File metaPath = new File(baseDumpDir, "binary_meta");
+        assertTrue(metaPath.exists());
+        assertTrue(metaPath.isDirectory());
+
+        assertEquals(1, metaPath.list().length);
+    }
+
+    /**
+     * Validates that {@code <baseDumpDir>/cp} directory exists and contains checkpoint history snapshot and markers.
+     */
+    private static void validateCpDir(File baseDumpDir) {
+        File cpPath = new File(baseDumpDir, "cp");
+        assertTrue(cpPath.exists());
+        assertTrue(cpPath.isDirectory());
+
+        assertThat(cpPath.list().length, greaterThanOrEqualTo(3));
+        assertTrue(new File(cpPath, CheckpointMarkersStorage.EARLIEST_CP_SNAPSHOT_FILE).exists());
+    }
+
+    /**
+     * Validates that {@code <baseDumpDir>/metastorage} directory exists and contains meta storage partition files.
+     */
+    private static void validateMetaStorageDir(File baseDumpDir) {
+        File metaStoragePath = new File(baseDumpDir, "metastorage");
+        assertTrue(metaStoragePath.exists());
+        assertTrue(metaStoragePath.isDirectory());
+
+        assertEquals(1, metaStoragePath.list().length);
+        // Somehow we use partition 0 instead of partition 1 for meta storage in this test, I don't know why.
+        String msPartFileName = String.format(PART_FILE_TEMPLATE, OLD_METASTORE_PARTITION);
+        assertTrue(new File(metaStoragePath, msPartFileName).exists());
+    }
+
+    /**
+     * Validates that {@code <baseDumpDir>/sys-cache} directory exists and contains all necessary cache files.
+     */
+    private static void validateUtilityCacheDir(File baseDumpDir) {
+        File utilityCachePath = new File(baseDumpDir, "sys-cache");
+        assertTrue(utilityCachePath.exists());
+        assertTrue(utilityCachePath.isDirectory());
+
+        // Also includes a single partition file.
+        assertEquals(3, utilityCachePath.list().length);
+
+        // cache_data.dat
+        assertTrue(new File(utilityCachePath, CACHE_DATA_FILENAME).exists());
+
+        // index.bin
+        assertTrue(new File(utilityCachePath, INDEX_FILE_NAME).exists());
```

Comment:
Why isn't partition file presence validated?

Reply:
I don't know the partition number. Should I scan the directory and check that there's a file that fits naming pattern?

Reply:
No, if it's too cumbersome/not too useful for the test

Reply:
Not really that useful